crates/nix_health/src/check/shell_configurations.rs

```diff
@@ -0,0 +1,55 @@
+use nix_rs::info;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, env, path::PathBuf};
+
+use crate::traits::{Check, CheckResult, Checkable};
+
+#[derive(Debug, Default, Serialize, Deserialize, Clone)]
+#[serde(default)]
+pub struct ShellConfigurations {}
+
+impl Checkable for ShellConfigurations {
+    fn check(&self, _: &info::NixInfo, _: Option<&nix_rs::flake::url::FlakeUrl>) -> Vec<Check> {
+        let check = Check {
+            title: "Shell Configurations".to_string(),
+            info: "are dotfiles managed by Nix ?".to_string(),
+            result: if check_shell_configuration() {
+                CheckResult::Green
+            } else {
+                CheckResult::Red {
+                    msg: "Shell Configurations are not managed by Nix".into(),
+                    suggestion: "Manage shell configurations through https://github.com/juspay/nixos-unified-template".into(),
+                }
+            },
+            required: false,
+        };
+        vec![check]
+    }
+}
+
+fn check_shell_configuration() -> bool {
+    // Define known dotfiles for different shells
+    let mut shell_dotfiles = HashMap::new();
+    shell_dotfiles.insert("zsh", vec![".zshrc"]);
+    shell_dotfiles.insert("bash", vec![".bashrc", ".bash_profile", ".profile"]);
+    // Check if shell configurations (zsh or bash) are managed by Nix
+    are_shell_dotfiles_nix_managed("zsh", &shell_dotfiles)
+        || are_shell_dotfiles_nix_managed("bashrc", &shell_dotfiles)
```

Comment:
The logic is wrong here. We want to check the current shell, and current shell only. 

Also, what happens if the user is using a different shell like `fish`?

Reply:
Also, 'tis time to refactor this ([encapsulate](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming))) by introducing a `struct` along with struct methods.

Reply:
> The logic is wrong here. We want to check the current shell, and current shell only.

@srid I tried using `SHELL` environment variable  for this but it turns out it gives the default shell of the user not the current active shell. There are other ways to determine the current active shell but those commands doesn't work well with rust. Hence I went ahead with this method for now.

![Screenshot 2024-10-09 at 7 04 15 PM](https://github.com/user-attachments/assets/a7c78d58-286b-4b33-9ecf-55aa3b58d331)

Reply:
@rsrohitsingh682 Do you understand why the logic is wrong? If I have both bash and zsh dotfiles, and bash dotfiles are managed by Nix but zsh dotfiles are not managed zsh - then what would be the result of the check if zsh is my current shell?

Reply:
oh! Right!! Got it. I made the check to use `OR` condition thinking what if user is just using only one shell either of (zsh or bash). But yeah I should have considered the case that you mentioned.